src/EnsembleKalmanProcesses.jl

```diff
@@ -46,89 +48,125 @@ Structure that is used in Ensemble Kalman processes
-     "vector of observations (length: N_data); mean of all observation samples"
-     obs_mean::Vector{FT}
-     "covariance of the observational noise, which is assumed to be normally distributed"
-     obs_noise_cov::Array{FT, 2}
-     "ensemble size"
-     N_ens::IT
-     "vector of arrays of size N_ensemble x N_data containing the model evaluations G(u) (in each EK iteration a new array of evaluations is added)"
-     g::Vector{Array{FT, 2}}
-     "vector of errors"
-     err::Vector{FT}
-     "vector of timesteps used in each EK iteration"
-     Δt::Vector{FT}
-     "the particular EK process (`Inversion` or `Sampler`)"
-     process::P
+    "Array of stores for parameters (u), each of size [N_data x N_ensemble]"
+    u::Array{DataContainer{FT}}
+    "vector of the observed vector size [N_data]"
+    obs_mean::Vector{FT}
+    "covariance of the observational noise, which is assumed to be normally distributed"
+    obs_noise_cov::Array{FT, 2}
+    "ensemble size"
+    N_ens::IT
+    "Array of stores for forward model outputs, each of size  [N_data x N_ensembles]; mean of all observation samples"
+    g::Array{DataContainer{FT}}
+    "vector of errors"
+    err::Vector{FT}
+    "vector of timesteps used in each EK iteration"
+    Δt::Vector{FT}
+    "the particular EK process (`Inversion` or `Sampler`)"
+    process::P
 end
 
 # outer constructors
-function EnsembleKalmanProcess(parameters::Array{FT, 2},
-               obs_mean,
-               obs_noise_cov::Array{FT, 2},
-               process::P;
-               Δt=FT(1)) where {FT<:AbstractFloat, P<:Process}
+function EnsembleKalmanProcess(params::Array{FT, 2},
+                               obs_mean,
+                               obs_noise_cov::Array{FT, 2},
+                               process::P;
+                               Δt=FT(1)) where {FT<:AbstractFloat, P<:Process}
 
+    init_params=DataContainer(params, data_are_columns=true)
     # ensemble size
-    N_ens = size(parameters)[1]
+    N_ens = size(get_data(init_params))[2] #stored with data as columns
     IT = typeof(N_ens)
     # parameters
-    u = Array{FT, 2}[] # array of Array{FT, 2}'s
-    push!(u, parameters) # insert parameters at end of array (in this case just 1st entry)
-    # G evaluations
-    g = Vector{FT}[]
+    g=[] #populated once we have evaluations
     # error store
     err = FT[]
     # timestep store
     Δt = Array([Δt])
 
-    EnsembleKalmanProcess{FT, IT, P}(u, obs_mean, obs_noise_cov, N_ens, g,
-                     err, Δt, process)
+    EnsembleKalmanProcess{FT, IT, P}([init_params], obs_mean, obs_noise_cov, N_ens, g,
+                                     err, Δt, process)
 end
 
 
+
+"""
+    get_X(ekp::EnsembleKalmanProcess, iteration::IT; return_array=true) where {IT <: Integer}
+
+Get X=u or X=g for the EKI iteration. Returns a DataContainer object unless array is specified.
+"""
+function get_u(ekp::EnsembleKalmanProcess, iteration::IT; return_array=true) where {IT <: Integer}
```

Comment:
We might want to have a getter for the whole sequence. E.g., `get_u(ekp)` that returns an array or list of Array{FT,2}.

Reply:
Done. I have implemented this to returns a list of DataContainers or a list of arrays. Along with some runtests.